File: src/main/frontend/src/payloads/Book.js
export default class Book {
    title
    description
    authorId

    constructor(title, description, authorId) {
        this.title = title
        this.description = description
        this.authorId = authorId
    }

    get title() {
        return this.title
    }

    get description() {
        return this.description
    }

    get authorId() {
        return this.authorId
    }

    set title(title) {
        this.title = title
    }
    
    set description(description) {
        this.description = description
    }

    set authorId(authorId) {
        this.authorId = authorId
    }
}

File: src/main/java/com/drdaza/CRUDGraphqlJPA/models/Book.java
package com.drdaza.CRUDGraphqlJPA.models;

import java.util.List;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.JoinTable;
import jakarta.persistence.ManyToMany;

@Entity(name = "books")
public class Book {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    @Column(name = "id_book")
    private Long id;

    @Column(nullable = false)
    private String title;
    private String description;

    @ManyToMany(fetch = FetchType.EAGER)
    @JoinTable(name = "book_has_authors", joinColumns = @JoinColumn(name = "book_id"), inverseJoinColumns = @JoinColumn(name = "author_id"))
    private List<Author> authors;

    public Book() {
    }

    public Book(String title, String description, List<Author> authors) {
        this.title = title;
        this.description = description;
        this.authors = authors;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public List<Author> getAuthors() {
        return authors;
    }

    public void setAuthors(List<Author> authors) {
        this.authors = authors;
    }
}
